### PR TITLE
Things

### PR DESCRIPTION
**Pull request by SandytheLizard**
1. Redid about 30% of the moves
2.Added a cli command called random_fight. this generates a random 6v6 battle. This command will crash if you have tuxemon in your party upon execution or if it picks a tuxemon without a valid shape.

3. Made it so stats revert after a battle.

### CODE REVIEW

File: mods/tuxemon/l18n/en_US/LC_MESSAGES/base.po [hunk 1]
```diff
@@ -170,22 +170,32 @@ msgid "combat_state_lifeleech_get"
...
 
 msgid "dodge_increased"
-msgstr "{user}'s {stat} increased."
+msgstr "{user 's {stat} increased."
+
+msgid "dodge_fell"
+msgstr "{user} 's dodge fell."
 
 msgid "speed_halved"
-msgstr "{user}'s speed was halved."
```
Comment (bitcraft) [inline mods/tuxemon/l18n/en_US/LC_MESSAGES/base.po:176]: please remove the space that was added here and in the next line

File: mods/tuxemon/l18n/en_US/LC_MESSAGES/base.po [hunk 2]
```diff
@@ -170,22 +170,32 @@ msgid "combat_state_lifeleech_get"
...
+msgstr "{user} 's hp was restored."
 
 msgid "combat_state_overfeed_get"
 msgstr "{user} Overfed {target}."
 
-msgid "dodge_fell"
-msgstr "{user}'s dodge fell."
 
 msgid "attacks_rose"
-msgstr "{user}'s attack damage types rose."
```
Comment (bitcraft) [inline mods/tuxemon/l18n/en_US/LC_MESSAGES/base.po:188]: please remove the space between the user/target and `'s`

File: tuxemon/event/actions/random_battle.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    npc_slug: str
+    tile_pos_x: int
+    tile_pos_y: int
+    animations: Union[str, None]
+    behavior: Union[str, None]
+
+
+logger = logging.getLogger(__name__)
+@final
+class randomfight(EventAction[AddMonsterActionParameters]):
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:48]: * class names must be CamelCase and begine with a capital letter
* class name in this file should match the name of the file.  ie: RandomBattle

File: tuxemon/event/actions/random_battle.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:21]: please run `black -l 79 tuxemon/event/actions/random_battle.py` to fix the formatting of this file

File: tuxemon/event/actions/random_battle.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    tile_pos_y: int
+    animations: Union[str, None]
+    behavior: Union[str, None]
+
+
+logger = logging.getLogger(__name__)
+@final
+class randomfight(EventAction[AddMonsterActionParameters]):
+    """
+    a
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:50]: Please be more descriptive.  and include what this does, specifically, and why it is difference than a "random encounter"

File: tuxemon/event/actions/random_battle.py [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+    """
+
+    name = "random_fight"
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:54]: please change fight to battle

File: tuxemon/event/actions/random_battle.py [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    param_class = AddMonsterActionParameters
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:55]: correct this

File: tuxemon/event/actions/random_battle.py [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+from tuxemon import ai
+from tuxemon.db import db
+
+from tuxemon.combat import check_battle_legal
+from tuxemon.event.eventaction import EventAction
+from typing import Union, NamedTuple, final, Optional
+from os import listdir
+from tuxemon.states.world.worldstate import WorldState
+from tuxemon.states.combat.combat import CombatState
+class AddMonsterActionParameters(NamedTuple):
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:36]: None of this is correct.  are parameters used?  if so, you need to rename this class and also change the values

File: tuxemon/event/actions/random_battle.py [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+    a
+
+    """
+
+    name = "random_fight"
+    param_class = AddMonsterActionParameters
+
+
+    def start(self) -> None:
+        file_path = 'tuxemon/db/monster/'
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:59]: it looks like you are trying to get all the available monster slugs.  if that is the case, there is no need to search the json from disk. you can get all the slugs for ``mon_slug_list` like this
```python
mon_slug_list = list(db.database["monster"].keys())
```
please remove all the code which reads the json and replace it with the code above.

File: tuxemon/event/actions/random_battle.py [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+            player.add_monster(current_monster)
+        
+        # Get a copy of the world state.
+        world = self.session.client.get_state_by_name(WorldState)
+
+        # Get the npc's parameters from the action
+        slug = '37707_male'
+
+        # Ensure that the NPC doesn't already exist on the map.
+        if slug in world.npcs:
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:88]: this creates a problem that this action will silently fail if the npc already exists.  please add an error logging statement that informs the player/developer that the battle cannot start.  or consider making a new NPC json that is only used for debugging battles like this one.

File: tuxemon/event/actions/random_battle.py [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+            return
+
+        # Get the npc's parameters from the action
+        pos_x = 0
+        pos_y = 0
+        behavior = 'wander'
+
+        sprite = False
+        if sprite:
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:97]: please remove all of this

File: tuxemon/event/actions/random_battle.py [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+            sprite = db.database["npc"][slug].get("sprite_name")
+
+        # Create a new NPC object
+        npc = tuxemon.npc.NPC(slug, sprite_name=sprite, world=world)
+        npc.set_position((pos_x, pos_y))
+
+        # Set the NPC object's variables
+        npc.behavior = behavior
+        npc.ai = ai.RandomAI()
+        npc.load_party()
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:113]: is this required?  you are just replacing the monsters in the next few lines

File: tuxemon/event/actions/random_battle.py [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+
+        for k in range(6):
+            monster_slug = random.choice(mon_slug_list)
+            current_monster = monster.Monster()
+            current_monster.load_from_db(monster_slug)
+            current_monster.set_level(5)
+            current_monster.current_hp = current_monster.hp
+            npc.add_monster(current_monster)
+        
+        # Don't start a battle if we don't even have monsters in our party yet.
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:123]: move this check to the top, before all the monster loading happens

File: tuxemon/event/actions/random_battle.py [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+        if not check_battle_legal(player):
+            logger.debug("battle is not legal, won't start")
+            return
+
+        world = self.session.client.get_state_by_name(WorldState)
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:128]: this is a duplicate

File: tuxemon/event/actions/random_battle.py [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+        if "environment" in player.game_variables:
+            env_slug = player.game_variables["environment"]
+        env = db.lookup(env_slug, table="environment")
+
+        # Add our players and setup combat
+        logger.debug("Starting battle!")
+        self.session.client.push_state(
+            CombatState,
+            players=(player, npc),
+            combat_type="player",
```
Comment (bitcraft) [inline tuxemon/event/actions/random_battle.py:142]: this is a `trainer` battle

**Review by bitcraft — CHANGES_REQUESTED** [review]
please see the comments

File: mods/tuxemon/l18n/en_US/LC_MESSAGES/base.po [hunk 16]
```diff
@@ -170,16 +170,16 @@ msgid "combat_state_lifeleech_get"
 msgstr "{user} is leeching health from {target}."
 
 msgid "dodge_increased"
-msgstr "{user 's {stat} increased."
+msgstr "{user's {stat} increased."
```
Comment (bitcraft) [inline mods/tuxemon/l18n/en_US/LC_MESSAGES/base.po]: should be `{user}`

File: mods/tuxemon/db/technique/avalanche.json [hunk 17]
```diff
@@ -3,14 +3,18 @@
...
+    "damage",
+    "statchange"
   ],
   "healing_power": 0,
   "icon": "",
   "is_area": true,
   "is_fast": false,
   "potency": 0.8,
   "power": 2.25,
+  "statspeedd": {
```
Comment (kerizane) [inline mods/tuxemon/db/technique/avalanche.json:15]: Typo?

**Review by kerizane — CHANGES_REQUESTED** [review]

**Pull request closed without merging**